src/main/java/com/aws/greengrass/ipc/GetClientDeviceAuthTokenOperationHandler.java

```diff
@@ -97,8 +102,12 @@ public GetClientDeviceAuthTokenResponse handleRequest(GetClientDeviceAuthTokenRe
                 String sessionId =
                         clientDevicesAuthServiceApi.getClientDeviceAuthToken(MQTT_CREDENTIAL_TYPE, credentialMap);
                 GetClientDeviceAuthTokenResponse response = new GetClientDeviceAuthTokenResponse();
+                domainEvents.emit(new GetClientDeviceAuthTokenEvent(GetClientDeviceAuthTokenEvent
```

Comment:
Following the logic from above, we'd then want to emit this at a lower layer. Moquette actually isn't using IPC. It's calling the service API directly. Ideally this should be coming from the session domain

Reply:
https://github.com/aws-greengrass/aws-greengrass-client-device-auth/blob/8682bcb81b23022fe43a95f18fe67229d8acbad8/src/main/java/com/aws/greengrass/clientdevices/auth/session/SessionManager.java#L66

You could catch it, emit the metric, then rethrow